client/lib/compare-props/test/index.js

```diff
@@ -0,0 +1,104 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import compareProps from '..';
+
+describe( 'compare-props', () => {
+	test( 'should do a shallow comparison when no options are supplied', () => {
+		const comparator = compareProps();
+
+		expect( comparator( { a: 1 }, { a: 1 } ) ).to.be.true;
+		expect( comparator( { a: 1, extra: 2 }, { a: 1 } ) ).to.be.false;
+		expect( comparator( { a: 1 }, { a: 1, extra: 2 } ) ).to.be.false;
+		expect( comparator( { a: [] }, { a: [] } ) ).to.be.false;
+		const a = { b: 1 };
+		expect( comparator( { a }, { a } ) ).to.be.true;
+	} );
+
+	test( 'should ignore properties marked as `ignore`', () => {
+		const comparator = compareProps( { ignore: [ 'irrelevant' ] } );
+
+		expect( comparator( { a: 1, irrelevant: 'whatever1' }, { a: 1, irrelevant: 'whatever2' } ) ).to
+			.be.true;
+
+		expect( comparator( { a: 1, irrelevant: 'whatever1' }, { a: 1 } ) ).to.be.true;
+
+		expect( comparator( { a: 1 }, { a: 1, irrelevant: 'whatever2' } ) ).to.be.true;
+	} );
+
+	test( 'should compare selected properties deeply', () => {
+		const comparator = compareProps( { deep: [ 'query' ] } );
+
+		expect(
+			comparator(
+				{
+					query: { text: 'plugin' },
+					page: 2,
+				},
+				{
+					query: { text: 'plugin' },
+					page: 2,
+				}
+			)
+		).to.be.true;
+
+		expect(
```

Comment:
This tests that everything not specified as deep stays as a shallow compare right?
Whats your take on cases like this where it feels like the unit test might be exceeding the bounds of the description?

The real description should be: 'should compare selected properties deeply while remaining properties are compared shallowly'

Reply:
I updated the test description.